drivers/sensor/grow_r502a/grow_r502a.c

```diff
@@ -0,0 +1,713 @@
+	if (ret == FPS_OK) {
+		LOG_INF("Correct password, FPS verified");
+	} else {
+		LOG_ERR("Package receive error 0x%X", ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int grow_r502a_sample_fetch(const struct device *dev,
+		enum sensor_channel chan)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+	const struct grow_r502a_config *cfg = dev->config;
+	int ret;
+
+	ret = fps_get_template_count(cfg->bus, &drv_data->template_count);
+	if (ret == FPS_OK) {
+		LOG_INF("Read success");
+	} else {
+		LOG_ERR("FPS template count get error 0x%X", ret);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int grow_r502a_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct grow_r502a_data *drv_data = dev->data;
+
+	if (chan == SENSOR_CHAN_FINGERPRINT) {
+		val->val1 = drv_data->template_count;
+	} else {
+		LOG_ERR("Invalid channel");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int grow_r502a_attr_set(const struct device *dev,
+			    enum sensor_channel chan,
+			    enum sensor_attribute attr,
+			    const struct sensor_value *val)
+{
+	const struct grow_r502a_config *cfg = dev->config;
+	int ret;
+
+	if (chan != SENSOR_CHAN_FINGERPRINT) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_ADD:
+		ret = fps_enroll(cfg->bus, val);
+		if (ret != 0) {
+			return ret;
+		}
+		break;
+	case SENSOR_ATTR_DEL:
+		ret = fps_delete(cfg->bus, val);
+		if (ret != 0) {
+			return ret;
+		}
+		break;
+	default:
+		LOG_ERR("Sensor attribute not supported");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int grow_r502a_attr_get(const struct device *dev,
+			    enum sensor_channel chan,
+			    enum sensor_attribute attr,
+			    struct sensor_value *val)
+{
+	const struct grow_r502a_config *cfg = dev->config;
+	int ret;
+
+	if (chan != SENSOR_CHAN_FINGERPRINT) {
+		LOG_ERR("Channel not supported");
+		return -ENOTSUP;
+	}
+
+	if (attr == SENSOR_ATTR_FIND) {
+		ret = fps_match(cfg->bus, val);
+		if (ret != 0) {
+			return ret;
+		}
+	} else {
+		LOG_ERR("attribute not supported");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static void grow_r502a_uart_flush(const struct device *dev)
+{
+	uint8_t c;
+
+	while (uart_fifo_read(dev, &c, 1) > 0) {
+		continue;
+	}
+}
+
+static int grow_r502a_init(const struct device *dev)
+{
+	const struct grow_r502a_config *cfg = dev->config;
+
+	if (!device_is_ready(cfg->bus)) {
+		LOG_ERR("%s: grow_r502a device not ready", dev->name);
+		return -ENODEV;
+	}
+
+#ifdef CONFIG_GROW_R502A_GPIO_POWER
+	if (!device_is_ready(cfg->vin_gpios.port)) {
+		LOG_ERR("GPIO port %s not ready", cfg->vin_gpios.port->name);
+		return -EINVAL;
+	}
+
+	int ret = gpio_pin_configure_dt(&cfg->vin_gpios, GPIO_OUTPUT_ACTIVE);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	gpio_pin_set_dt(&cfg->vin_gpios, 0);
+	k_sleep(K_MSEC(100));
+
+	if (!device_is_ready(cfg->act_gpios.port)) {
+		LOG_ERR("GPIO port %s not ready", cfg->act_gpios.port->name);
+		return -EINVAL;
```

Comment:
`-ENODEV`

Reply:
Thanks for the review, Solved this and all other changes requested on previous comments.